ksql-engine/src/test/java/io/confluent/ksql/analyzer/ExpressionAnalyzerTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.analyzer;
+
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableSet;
+import io.confluent.ksql.execution.expression.tree.ComparisonExpression;
+import io.confluent.ksql.execution.expression.tree.ComparisonExpression.Type;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.QualifiedName;
+import io.confluent.ksql.execution.expression.tree.QualifiedNameReference;
+import io.confluent.ksql.execution.expression.tree.StringLiteral;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.SchemaUtil;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ExpressionAnalyzerTest {
+
+  private static final Expression WINDOW_START_EXP = new QualifiedNameReference(
+      QualifiedName.of("something", SchemaUtil.WINDOWSTART_NAME)
+  );
+
+  private static final Expression OTHER_EXP = new StringLiteral("foo");
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  @Mock
+  private SourceSchemas sourceSchemas;
+  private ExpressionAnalyzer analyzer;
+
+  @Before
+  public void setUp() {
+    analyzer = new ExpressionAnalyzer(sourceSchemas);
+  }
+
+  @Test
+  public void shouldNotThrowOnWindowStartIfAllowed() {
+    // Given:
+    final Expression expression = new ComparisonExpression(
+        Type.EQUAL,
+        WINDOW_START_EXP,
+        OTHER_EXP
+    );
+
+    // When:
+    analyzer.analyzeExpression(expression, true);
+
+    // Then: did not throw
```

Comment:
The test will pass if analyzer.analyzeExpression does nothing at all. Is there some result that can be verified?

Reply:
Nope. The only thing the class under test does is validation, i.e. it either throws or it doesn't. This test tests it doesn't throw for a specific use case